### PR TITLE
added hkt doorSensor & peopleCounter

### PR DESCRIPTION
**Pull request by J2thatsme**

### CODE REVIEW

File: types/hkt/doorSensor/meta.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "Door Sensor",
```
Comment (mariusgiger) [inline types/hkt/doorSensor/meta.json:2]: DS20?

File: types/hkt/doorSensor/system.schema.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      "description": "Counting interval",
+      "hideFromKpis": true
+    },
+    "lifecycleInterval": {
+      "title": "Lifecycle interval",
+      "type": "integer",
+      "description": "Lifecycle interval",
+      "hideFromKpis": true
+    },
+    "instaled": {
```
Comment (mariusgiger) [inline types/hkt/doorSensor/system.schema.json:43]: typo installed

File: types/hkt/doorSensor/system.schema.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    "swVersion": {
+      "title": "Software version",
+      "type": "integer",
+      "description": "Software version",
+      "hideFromKpis": true
+    },
+    "Working mode": {
+      "title": "Mode",
+      "type": "integer",
+      "description": "Working mode 1: During the counting hours, the device will periodically report the number of times the door sensor is openedand closed in the previous cycle and the total accumulatedtimes. Outside the counting hours, the device runs in trigger mode. Working mode 2: Trigger mode (Default): Every time the tamper alarm is triggered or the door is opened or closed, the devicereportsdata immediately.",
```
Comment (mariusgiger) [inline types/hkt/doorSensor/system.schema.json:22]: Working mode 1: During the counting hours, the device will periodically report the number of times the door sensor is opened and closed in the previous cycle and the total accumulated times. Outside the counting hours, the device runs in trigger mode. Working mode 2: Trigger mode (Default): Every time the tamper alarm is triggered or the door is opened or closed, the device reports data immediately.

File: types/hkt/doorSensor/uplink.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        door.absOpenings = Bits.bitsToUnsigned(
+          bits.substr((pointer += 16), 32),
+        );
+        door.absClosings = Bits.bitsToUnsigned(
+          bits.substr((pointer += 32), 32),
+        );
+        pointer += 32;
+        break;
+      case 0x84:
+        system.instaled = !!Bits.bitsToUnsigned(bits.substr(48, 8));
```
Comment (mariusgiger) [inline types/hkt/doorSensor/uplink.js:77]: installed

File: types/hkt/peopleCounter/default.schema.json [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      "type": "integer",
+      "description": "Counter A"
+    },
+    "counterB": {
+      "title": "Counter B",
+      "type": "integer",
+      "description": "Counter B"
+    },
+    "absCountA": {
+      "title": "Absoulut count A",
```
Comment (mariusgiger) [inline types/hkt/peopleCounter/default.schema.json:18]: Absolute

File: types/hkt/peopleCounter/default.schema.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      "type": "integer",
+      "description": "Absoulut count A"
+    },
+    "absCountB": {
+      "title": "Absoulut count B",
```
Comment (mariusgiger) [inline types/hkt/peopleCounter/default.schema.json:23]: Absolute

File: types/hkt/peopleCounter/meta.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "People Counter",
+  "version": "1.0.0",
+  "manufacturer": "HKT",
+  "url": "https://www.hkttech.com/cn/LoRa-People-Counter/",
+  "description": "The LoRa People Counter is a battery powered sensor for counting people crossing a horizontal line. The counter consists of two parts, an infrared (IR) transmitter part and an infrared receiver part, which always work together. The IR transmitter the infrared signal.",
```
Comment (mariusgiger) [inline types/hkt/peopleCounter/meta.json:6]: What does this mean? The IR transmitter the infrared signal.?

File: types/hkt/peopleCounter/system.schema.json [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+      "description": "Threshold of accumulative number of counts",
+      "hideFromKpis": true
+    },
+    "noInfrared": {
+      "title": "Infrared error",
+      "type": "boolean",
+      "description": "Infrared error",
+      "hideFromKpis": true
+    },
+    "instaled": {
```
Comment (mariusgiger) [inline types/hkt/peopleCounter/system.schema.json:44]: installed

File: types/hkt/peopleCounter/uplink.js [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+      case 0x86:
+        pointer += 24;
+        break;
+      default:
+        pointer = bits.length;
+        break;
+    }
+  }
+
+  if (Object.keys(data).length > 0) {
```
Comment (mariusgiger) [inline types/hkt/peopleCounter/uplink.js:85]: if you directly emit the messages in the switch this keys check is not required

**Review by mariusgiger — CHANGES_REQUESTED** [review]
a few proposals

File: types/hkt/peopleCounter/meta.json (hunk 7 above)
Comment (J2thatsme) [reply types/hkt/peopleCounter/meta.json:6]: Copied from the website but i changed it now ^^

**Review by mariusgiger — APPROVED** [review]

**Review by simonrieser — APPROVED** [review]

**Pull request merged**